core/src/oauth/providers/zendesk.rs

```diff
@@ -45,8 +48,20 @@ impl Provider for ZendeskConnectionProvider {
             "scope": "read"
         });
 
+        let sub_domain = match connection.metadata()["extra_config"].as_str() {
```

Comment:
Shouldn't we define `extra_config` as an object to support more in the future? Which seems very likely something we will need.

Reply:
Well technically metadata is a Value in `oauth` so we definitely could, but at the same time we won't get much guarantee on the typing through `oauth` so that would be a pure `front`/`connectors` convention.

Reply:
The whole point of metadata is to be that very object so I'm bearish of adding an object in an object. I can refactor the code to return an extraConfig Record<string, string> at setup that gets flattened in the metadata so that core will store `zendesk_subdomain` in metadata instead of an opaque `extra_config`. I think that's the best approach, wdyt?

Reply:
Yes as long as it's clearer what `extra_config` holds I'm fine.

Reply:
extra_config will just live on front during the connection setup. The problem is that whatever we build we will have to pass as GET parameter.... I presume we can just stringify (maybe base64 encode) it.